allauth/socialaccount/providers/microsoft/provider.py

```diff
@@ -6,7 +6,10 @@
 
 class MicrosoftGraphAccount(ProviderAccount):
     def get_avatar_url(self):
-        return self.account.extra_data.get("photo")
+        return "data:{format};base64, {data}".format(
+            format=self.account.extra_data.get('photo_metadata', {}).get('@odata.mediaContentType', 'image/jpeg'),
+            data=self.account.extra_data.get('photo')
```

Comment:
**Issue**: I would expect that if `extra_data` does not contain `photo_metadata` or `photo`, then `get_avatar_url()` would return `None`, but in this case it returns a malformatted `data:...` string.

Reply:
Yes this seems super obvious upon reflection so I apologise, it's an easy change to make.

Reply:
from __future__ import unicode_literals
from allauth.socialaccount.providers.base import AuthAction, ProviderAccount
from allauth.socialaccount.providers.oauth2.provider import OAuth2Provider

class MicrosoftGraphAccount(ProviderAccount):
    def get_avatar_url(self):
        return self.account.extra_data.get("photo")
        return "data:{format};base64, {data}".format(
            format=self.account.extra_data.get('photo_metadata', {}).get('@odata.mediaContentType', 'image/jpeg'),
            data=self.account.extra_data.get('photo')
Owner
@[pennersr](https://github.com/pennersr) pennersr 2 days ago
Issue: I would expect that if extra_data does not contain photo_metadata or photo, then get_avatar_url() would return None, but in this case it returns a malformatted data:... string.

Author
@[arowley-ai](https://github.com/arowley-ai) arowley-ai yesterday
Yes this seems super obvious upon reflection so I apologise, it's an easy change to make.